### DISCUSSION TITLE
[Feature Request] Adding isSubmitting Prop to the Button

### DISCUSSION THREAD

**Discussion by kimskovhusandersen** [heroui-inc/heroui#4724] [Feature requests]
### Is your feature request related to a problem? Please describe.

Hey,
First of all, thanks a lot for all the amazing work you've done with NextUI! I've really enjoyed using it, and I'm looking forward to seeing it get even better.

I've come across a problem with our button component that might affect users, especially those who rely on screen readers and keyboards. When the button is loading, it loses focus and can't be interacted with, which can be quite confusing.

### Describe the solution you'd like

**Introduce 'isSubmitting' Property.**
Instead of using the regular disabled attribute, we would use aria-disabled=true. This change would keep the focus on the button during submission, which helps to prevent the focus from jumping around and allows users to still see tooltips for more information. To handle issues like double-clicks or accidental re-submissions, we'd use e.preventDefault() in our onClick or onPress handlers. For the visual part, we’d keep the disabled style and keep using a spinner or some indicator to show it’s still pending. This way, users can still interact with it enough to get tooltips and focus would stay on the button, which would be really helpful.

### Describe alternatives you've considered

Or, we could slightly change how the isLoading works. This approach might be a bit tricky as it could introduce breaking changes. The idea is the same as for the first proposal: to allow the button to still show tooltips or messages even while it’s loading. Like in the first approach, we'd use aria-disabled="true", and adjust the click behavior so it doesn’t trigger the main action but allows for other interactions like tooltips.

**Comment by linear** [comment]
<p><a href="https://linear.app/nextui-inc/issue/ENG-1508/[feature-request]-adding-issubmitting-prop-to-the-button">ENG-1508 [Feature Request] Adding isSubmitting Prop to the Button</a></p>

**Comment by Fahad-Dezloper** [comment]
Hi! I would like to work on this issue. Could you please assign it to me?
